packages/phoenix-event-display/src/tests/managers/loading-manager.test.ts

```diff
@@ -0,0 +1,54 @@
+import { LoadingManager } from '../../managers/loading-manager';
+
+describe('LoadingManager', () => {
+  let loadingManager: LoadingManager;
+
+  beforeEach(() => {
+    loadingManager = new LoadingManager();
+  });
+
+  afterEach(() => {
+    loadingManager.reset();
+  });
+
+  it('should create an instance', () => {
+    expect(loadingManager).toBeTruthy();
+  });
+
+  it('should add an item which is to be loaded', () => {
+    loadingManager.addLoadableItem('item');
+    expect(loadingManager.toLoad.length).toBe(1);
+  });
+
+  it('should call the function when an item has finished loading', () => {
+    jest.spyOn(loadingManager, 'itemLoaded');
+    loadingManager.addLoadableItem('item');
+    loadingManager.itemLoaded('item');
+    expect(loadingManager.itemLoaded).toHaveBeenCalled();
```

Comment:
This is redundant. The correct way to assert that the loading was finished would be to add a load listener (`jest.fn()`) using `addLoadListener` and make sure that the load listener/callback was called.

Reply:
Here's how I would test it:

```ts
it('should call the listeners when all items have loaded', () => {
  const callback = jest.fn();
  loadManager.addLoadListener(callback);
  loadingManager.addLoadableItem('item');
  loadingManager.itemLoaded('item');
  expect(callback).toHaveBeenCalled();
});
```

Reply:
Thanks! Right ✅